examples/svdiff/train_svdiff.py

```diff
@@ -0,0 +1,1121 @@
+                    target = noise_scheduler.get_velocity(latents, noise, timesteps)
+                else:
+                    raise ValueError(f"Unknown prediction type {noise_scheduler.config.prediction_type}")
+
+                if args.with_prior_preservation:
+                    # Chunk the noise and model_pred into two parts and compute the loss on each part separately.
+                    model_pred, model_pred_prior = torch.chunk(model_pred, 2, dim=0)
+                    target, target_prior = torch.chunk(target, 2, dim=0)
+
+                    # Compute instance loss
+                    loss = F.mse_loss(model_pred.float(), target.float(), reduction="mean")
+
+                    # Compute prior loss
+                    prior_loss = F.mse_loss(model_pred_prior.float(), target_prior.float(), reduction="mean")
+
+                    # Add the prior loss to the instance loss.
+                    loss = loss + args.prior_loss_weight * prior_loss
+                else:
+                    loss = F.mse_loss(model_pred.float(), target.float(), reduction="mean")
+
+                accelerator.backward(loss)
+                if accelerator.sync_gradients:
+                    params_to_clip = (
+                        itertools.chain(unet.parameters(), text_encoder.parameters())
+                        if args.train_text_encoder
+                        else unet.parameters()
+                    )
+                    accelerator.clip_grad_norm_(params_to_clip, args.max_grad_norm)
+                optimizer.step()
+                lr_scheduler.step()
+                optimizer.zero_grad(set_to_none=args.set_grads_to_none)
+
+            # Checks if the accelerator has performed an optimization step behind the scenes
+            if accelerator.sync_gradients:
+                progress_bar.update(1)
+                global_step += 1
+
+                if accelerator.is_main_process:
+                    if global_step % args.checkpointing_steps == 0:
+                        save_path = os.path.join(args.output_dir, f"checkpoint-{global_step}")
+                        save_weights(global_step, save_path)
+                        # TODO: support accelerator saving
+                        # accelerator.save_state(save_path)
+                        # logger.info(f"Saved state to {save_path}")
+
+                    if args.validation_prompt is not None and global_step % args.validation_steps == 0:
+                        log_validation(text_encoder, tokenizer, unet, vae, args, accelerator, weight_dtype, epoch)
+
+            logs = {"loss": loss.detach().item(), "lr": lr_scheduler.get_last_lr()[0]}
+            progress_bar.set_postfix(**logs)
+            accelerator.log(logs, step=global_step)
+
+            if global_step >= args.max_train_steps:
+                break
+    # put the last checkpoint to output-dir
+    save_weights(global_step, save_path=args.output_dir)
```

Comment:
I think this needs to go under `accelerator.is_main_process()` following other examples like:

https://github.com/huggingface/diffusers/blob/ce5504934ac484fca39a1a5434ecfae09eabdf41/examples/dreambooth/train_dreambooth.py#L1331

Reply:
@sayakpaul actually, it's applied inside the func `save_weights`. 
https://github.com/huggingface/diffusers/blob/8954abe8ee9fb71efee445ab21c88489a7fb63a8/examples/svdiff/train_svdiff.py#L908
https://github.com/huggingface/diffusers/blob/8954abe8ee9fb71efee445ab21c88489a7fb63a8/examples/svdiff/train_svdiff.py#L910

Reply:
Ah okay. I am fine with it.

Reply:
let me resolve this conversation then.